clang/lib/AST/Interp/IntegralAP.h

```diff
@@ -249,6 +248,11 @@ template <bool Signed> class IntegralAP final {
     R->V = A.V - B.V;
     return false; // Success!
   }
+
```

Comment:
Is this template forward declared?  Either way, I think as none of the calls are dependent, this ends up being UB (As the lookup 'changes' between the declaration and the call).

All that to say: I think you just have to move this definition above its uses.